python/ray/train/predictor.py

```diff
@@ -76,6 +76,16 @@ class Predictor(abc.ABC):
     def __init__(self, preprocessor: Optional[Preprocessor] = None):
         """Subclasseses must call Predictor.__init__() to set a preprocessor."""
         self._preprocessor: Optional[Preprocessor] = preprocessor
+        # Whether tensor columns should be automatically cast from/to the tensor
+        # extension type at UDF boundaries. This can be overridden by subclasses.
+        self._cast_tensor_columns = False
+
+    @DeveloperAPI
```

Comment:
Should this be completely private?

I don't think developers implementing their own Predictors need to know about this.

Reply:
I suppose that we don't support developers extending the `BatchPredictor` or providing their own batch prediction engine, right? If so, I'll make this private.

Reply:
Yep that's right...I doubt that would be a use case